network/src/request_manager.rs

```diff
@@ -0,0 +1,379 @@
+// Copyright (C) 2019-2020 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::external::{message::message::Message, message_types::Version, Channel, Handshake, HandshakeStatus, Verack};
+
+use std::{
+    collections::{HashMap, HashSet},
+    net::SocketAddr,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+};
+use tokio::{net::TcpStream, sync::RwLock, task};
+
+#[derive(Debug, Clone)]
+pub struct RequestManager {
+    /// The handshakes with connected peers
+    handshakes: Arc<RwLock<HashMap<SocketAddr, Handshake>>>,
+    /// A list of remote addresses currently sending a request.
+    pending_addresses: Arc<RwLock<HashSet<SocketAddr>>>,
+    /// A counter for the number of send requests the manager processes.
+    send_request_count: Arc<AtomicU64>,
+    /// A counter for the number of send requests that succeeded.
+    send_success_count: Arc<AtomicU64>,
+    /// A counter for the number of send requests that failed.
+    send_failure_count: Arc<AtomicU64>,
+}
+
+impl RequestManager {
+    /// Creates a new instance of a `ConnectionManager`.
+    #[inline]
+    pub fn new() -> Self {
```

Comment:
`RequestManager` should instead just `#[derive(Default)]`, which would create an object with the same values as listed below